client/lib/media/utils.js

```diff
@@ -299,15 +299,27 @@ const MediaUtils = {
 
 	/**
 	 * Returns true if the specified item exceeds the maximum upload size for
-	 * the given site. Returns null if the bytes are invalid or the max upload
-	 * size for the site is unknown. Otherwise, returns true.
+	 * the given site. Returns null if the bytes are invalid, the max upload
+	 * size for the site is unknown or a video is being uploaded for a Jetpack
+	 * site with VideoPress enabled. Otherwise, returns true.
 	 *
-	 * @param  {Number}   bytes A file size to check, as bytes
+	 * @param  {Object}   item  Media object
 	 * @param  {Object}   site  Site object
 	 * @return {?Boolean}       Whether the size exceeds the site maximum
 	 */
-	isExceedingSiteMaxUploadSize: function( bytes, site ) {
-		if ( ! isFinite( bytes ) || ! site || ! site.options || ! site.options.max_upload_size ) {
+	isExceedingSiteMaxUploadSize: function( item, site ) {
+		const bytes = item.size;
+
+		if ( ! site || ! site.options ) {
+			return null;
+		}
+
+		if ( ! isFinite( bytes ) || ! site.options.max_upload_size ) {
+			return null;
+		}
+
+		if ( site.jetpack && site.isModuleActive( 'videopress' ) && versionCompare( site.options.jetpack_version, '4.5', '>=' ) &&
```

Comment:
A few notes:

- I'm not really familiar with `isSupportedFileTypeInPremium`. Is that enough to ensure that the file is a video, or at the very least one which can ignore file size limits?
- When splitting condition on multiple lines, we should indent following lines ([related guideline](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#multi-line-statements))
- The `JetpackSite` prototype [has a `versionCompare` method](https://github.com/Automattic/wp-calypso/blob/f0a9ec7e7bea377e1c1d4be153b411f7bc8eed8f/client/lib/site/jetpack.js#L44-L46) you can use as a convenience